### ISSUE TITLE
Ansible-lint mistakenly believes that a free strategy is being used

### ISSUE THREAD

**Issue by kopytov** [ansible/ansible-lint#3331]
##### Summary

Ansible-lint mistakenly thinks that the free strategy is used, although the linear strategy is used. And because of this it warns about problems on tasks with run_once when it shouldn't.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
```
```
ansible-lint 6.15.0 using ansible 2.14.4
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

```
cat > playbook.yaml <<EOF
---
- name: Get network info
  hosts: all
  strategy: linear
  gather_facts: false
  tasks:
    - name: Get list of available networks
      delegate_to: localhost
      run_once: true
      openstack.cloud.networks_info:
      register: networks
EOF
ansible-lint playbook.yaml
```

```console (paste below)
WARNING  Listing 1 violation(s) that are fatal
run-once[task]: Using run_once may behave differently if strategy is set to free.
playbook.yaml:7 Task/Handler: Get list of available networks

Read documentation for instructions on how to ignore specific rule violations.

              Rule Violation Summary
 count tag            profile rule associated tags
     1 run-once[task]         idiom

Failed after production profile, 5/5 star rating: 1 failure(s), 0 warning(s) on 1 files.
```

##### Desired Behavior

Ansible-lint should not report a problem in this case.

##### Actual Behavior

Ansible-lint believes that a free strategy is being used and report a problem.

**Comment by ajinkyau** [comment]
@kopytov Please read the documentation and warning mentioned in the docs carefully.
https://ansible-lint.readthedocs.io/rules/run-once/
